### PR TITLE
e2e: Fixups for conformance template generation

### PR DESCRIPTION
**Pull request by randomvariable**
Resolves a CI difference where _artifacts is not the same as the ARTIFACTS directory, by moving the logic from GNU Make to go

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,7 +47,8 @@ E2E_DATA_DIR ?= $(REPO_ROOT)/test/e2e_new/data
 E2E_CONF_PATH  ?= $(E2E_DATA_DIR)/e2e_conf.yaml
 KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
 KUBETEST_FAST_CONF_PATH ?= $(abspath $(REPO_ROOT)/test/e2e_new/data/kubetest/conformance-fast.yaml)
-CONFORMANCE_CI_TEMPLATE := $(ARTIFACTS)/templates/cluster-template-conformance-ci-artifacts.yaml
+CONFORMANCE_CI_TEMPLATE := _artifacts/templates/cluster-template-conformance-ci-artifacts.yaml
```
Comment (detiber) [inline Makefile:50]: Is this related to something hardcoded in a kustomize file? I'm wondering if using envsubst or similar might be better than having a hardcoded path to _artifacts

Comment (detiber) [reply Makefile:50]: Maybe something we could add to the e2e config parser?

Comment (randomvariable) [reply Makefile:50]: Let me check

Comment (randomvariable) [reply Makefile:50]: TBH, don't like this much logic in the Makefile as it prohibits `go test`

Comment (detiber) [reply Makefile:50]: Yeah, that's one of the reasons I was thinking that maybe we need to push that support into the e2e config parser

Comment (randomvariable) [reply Makefile:50]: modifying at run time in go, but github is not showing the changes...

**Comment by randomvariable** [comment]
Pod never came up for test
/test pull-cluster-api-provider-aws-make-conformance

**Comment by randomvariable** [comment]
hmm, looks like github is causing problems.

File: .gitignore [hunk 2]
```diff
@@ -58,3 +58,5 @@ junit.*.xml
 
 # test results
 _artifacts
+awsiamconfiguration.yaml
```
Comment (gab-satchi) [inline .gitignore:61]: do we need these? thought they also get created under the artifacts folder

File: test/e2e_new/conformance_test.go [hunk 3]
```diff
@@ -99,3 +105,56 @@ var _ = Describe("conformance tests", func() {
 	})
 
 })
+
+func prepConformanceConfig(artifactsDir string, e2eConfig *clusterctl.E2EConfig) error {
+	for i, p := range e2eConfig.Providers {
+		if p.Name != "aws" {
+			continue
+		}
+		e2eConfig.Providers[i].Files = append(p.Files, clusterctl.Files{
```
Comment (gab-satchi) [inline test/e2e_new/conformance_test.go:114]: we could also leave the copying and kustomize call in the Makefile and then append the file here after ensuring it exists.

File: .gitignore (hunk 2 above)
Comment (randomvariable) [reply .gitignore:61]: Haven't got to the bottom as to why, but sometimes it gets created in the test directory. Will open a follow up issue to resolve than try to solve it now.

File: test/e2e_new/conformance_test.go (hunk 3 above)
Comment (randomvariable) [reply test/e2e_new/conformance_test.go:114]: Idea is to start moving towards not needing the makefile, and allowing `go test` to work if you wanted to use it that way.

File: .gitignore (hunk 2 above)
Comment (randomvariable) [reply .gitignore:61]: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1827

File: test/e2e_new/conformance_test.go [hunk 4]
```diff
@@ -99,3 +105,56 @@ var _ = Describe("conformance tests", func() {
...
+			SourcePath: path.Join(artifactFolder, "/templates/cluster-template-conformance-ci-artifacts.yaml"),
+			TargetName: "cluster-template-conformance-ci-artifacts.yaml",
```
Comment (detiber) [inline test/e2e_new/conformance_test.go:116]: Is there no way currently to trigger a conformance test against a released version of Kubernetes? It looks like everything is hard coded towards running against ci artifacts.

Comment (randomvariable) [reply test/e2e_new/conformance_test.go:116]: We always generate the conformance template, but https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/test/e2e_new/conformance_test.go#L59 determines whether or not it actually gets used.

**Pull request merged**